pyro/infer/abstract_infer.py

```diff
@@ -72,10 +72,15 @@ def _gen_weighted_samples(self, *args, **kwargs):
         raise NotImplementedError("_gen_weighted_samples is abstract method")
 
     def sample(self, *args, **kwargs):
+        sample_shape = kwargs.pop("sample_shape", None)
+        if sample_shape:
+            raise ValueError("Arbitrary `sample_shape` not supported by Distribution class.")
```

Comment:
nit: replace `Distribution` with `Histrogram`, right?

Reply:
I meant Distribution subclass so that I could use it uniformly in other places..but its better to be explicit. Will change to actual class names.

Reply:
But `Distribution`s do support arbitrary sample shape, don't they?

Reply:
Yes, Distributions do; meant this particular Distribution sub-class does not. :) Never mind, changed to actual class names. We should fix the `sample_shape` in Rejectors separately.